docs/rhoas/iac_tools_rhoas/README.adoc

```diff
@@ -0,0 +1,680 @@
+. Copy the contents of the `create_kafka_acl_binding` module and paste it into the `rhoas_kafka.yml` file to create an ACL for the service account with some default values, and bind that ACL to the Kafka instance and service account.
+. You can allow Ansible to get the Kafka ID from the `kafka_req_resp` variable or enter it in the `kafka_id` field.
++
+The following list is a description of each field that must have a value in an ACL binding module.
+
+`principal`:: The user that this binding applies to. This example uses the service account client ID.
+`resource_name`:: The name of the resource you want access to. This example uses the name that is passed when creating the topic.
+`resource_type`:: The type of resource you want access to. This example uses *Topic*.
+`pattern_type`:: The type of pattern of the ACL. The below example uses the `PREFIXED` pattern type meaning that Kafka will try to match the prefix of the resource name with the resource specified in the ACL.
+`operation_type`:: The type of operation (an action performed on a resource) that is allowed for the given user on this module.
+`permission_type`:: Whether permission is given to the user or taken away.
++
+.Example `create_kafka_acl_binding` module
+[source,yaml]
+----
+- name: Create kafka ACL Service Binding
+    rhoas.rhoas.create_kafka_acl_binding:
+      kafka_id: "{{ kafka_req_resp.kafka_id }}"
+      # To hard code the kafka_id, uncomment and use the following line:
+      # kafka_id: "KAFKA_ID"
+      principal: " {{ srvce_acc_resp_obj['client_id'] }}"
+      # To hard code the principal_id, uncomment and use the following line:
+      # principal: "PRINCIPAL_ID"
+      resource_name: "topic-name"
+      resource_type: "Topic"
+      pattern_type: "PREFIXED"
+      operation_type: "all"
+      permission_type: "allow"
+      openshift_offline_token: "OPENSHIFT_OFFLINE_TOKEN"
+    register: kafka_acl_resp
+
+----
+
+. Copy the contents of the `create_kafka_topic` module and paste it into the `rhoas_kafka.yml` file to create a Kafka topic.
+. You can allow Ansible to get the Kafka ID  from the `kafka_req_resp.id` variable or enter it in the `kafka_id` field.
+
++
+.Example `create_kafka_topic` module
+[source,yaml]
+----
+- name: Create Kafka Topic
+    create_kafka_topic:
+      topic_name: "kafka-topic-name"
+      kafka_id: "{{ kafka_req_resp.id }}"
+      # To hard code the kafka_id, uncomment and use the following line:
+      # kafka_id: "KAFKA_ID"
+      partitions: 1
+      retention_period_ms: "86400000"
+      retention_size_bytes: "1073741824"
+      cleanup_policy: "compact"
+      openshift_offline_token: "OPENSHIFT_OFFLINE_TOKEN"
+    register:
+      create_topic_res_obj
+----
++
+. Copy the `update_kafka_topic` module and paste it into the `rhoas_kafka` file to update the configuration of the topic. In the following example, the cleanup policy has been updated from compact to delete by replacing `"compact"` with `"delete"` in the `cleanup_policy` field.
```

Comment:
For the remainder of the Ansible example, I suggest continuing to follow this pattern:

. To create a module that does <task>, copy the <module_name> module from the example playbook and paste it in your `rhoas_kafka.yml` file`. The <module-name> module from the example playbook is shown below:
+
[code block]
. Things you must (or can) specify.
+
Result.
```

Reply:
Hmmm, maybe. But we don't really make any reference to these example code blocks, so I found them to be kind of... _free-floating_.

Maybe a slightly less formal reference along the following lines would work?

```
 To create a module that does <task>, copy the <module_name> module shown below from the example playbook.
 ```

Reply:
WDYT?

Reply:
Yeah that works for me. :)